src/pf/dispatch.lua

```diff
@@ -0,0 +1,164 @@
+module(...,package.seeall)
+
+---
+--- Program := Captures '=>' Consequent
+--- Captures := '(' [ Identifier' [',' Identifier]... ] ')'
+--- Consequent := Call | Dispatch
+--- Call := Identifier '(' [ ArithmeticExpression [ ',' ArithmeticExpression ] ] ')'
+--- Dispatch := '{' Clause... '}'
+--- Clause := LogicalExpression '=>' Consequent
+---
+--- LogicalExpression and ArithmeticExpression are embedded productions
+--- of pflang.
+---
+
+local utils = require('pf.utils')
+local set, assert_equals = utils.set, utils.assert_equals
+local parse_pflang = require('pf.parse').parse
+
+local function split(str, pat)
+   pat = '()'..pat..'()'
+   local ret, start_pos = {}, 1
+   local tok_pos, end_pos = str:match(pat)
+   while tok_pos do
+      table.insert(ret, str:sub(start_pos, tok_pos - 1))
+      start_pos = end_pos
+      tok_pos, end_pos = str:match(pat, start_pos)
+   end
+   table.insert(ret, str:sub(start_pos))
+   return ret
+end
+
+local function scanner(str)
+   local pos = 1
+   local function error_str(message, ...)
+      local location_error_message = "Parse error: In expression \"%s\""
```

Comment:
Please make this more explicit, ie by saying "Pfmatch parse error". We've already had an issue filed when it wasn't clear what layer an error message was coming from.